internal/backend/local/backend_local.go

```diff
@@ -77,7 +77,12 @@ func (b *Local) localRun(op *backend.Operation) (*backend.LocalRun, *configload.
 
 	var ctxDiags tfdiags.Diagnostics
 	var configSnap *configload.Snapshot
-	if op.PlanFile != nil {
+	if op.PlanFile.IsCloud() {
+		diags = diags.Append(fmt.Errorf("error: using a saved cloud plan with the local backend is not supported"))
```

Comment:
I think we'll want to reword this error message a bit since this local backend logic is shared by _all_ the backends that execute locally, so it will be somewhat confusing if you're not using the literal `local` backend! For instance, if you're using TFC for state storage only (that is, using the `cloud` integration with a workspace in local execution mode) and attempt to apply a saved cloud plan, you get this error.

We don't have an easy way to get the name of the actual resolved backend at this point, so maybe something like this?

```suggestion
		diags = diags.Append(fmt.Errorf("error: using a saved cloud plan when executing Terraform locally is not supported"))
```